osgar/drivers/kloubak.py

```diff
@@ -0,0 +1,283 @@
+#                self.last_encoders_front_left = None
+#                self.last_encoders_front_right = None
+#                self.last_encoders_rear_left = None
+#                self.last_encoders_rear_right = None
+                return True
+        return False
+
+    def slot_can(self, timestamp, data):
+        self.time = timestamp
+        limit_brake = 100
+        if abs(self.desired_angular_speed) < math.radians(5):
+            limit_r = 400
+            limit_l = 400
+        elif self.desired_angular_speed > 0:  # TODO handle backup
+            # turn left
+            limit_r = 400
+            limit_l = 200
+        else:
+            # turn right
+            limit_r = 200
+            limit_l = 400
+        if self.process_packet(data):
+#            print(self.last_encoders_front_left, self.last_encoders_front_right)
+            fwd = [0, 0, 72, 0]  # 6Amp
+            bwd = [255, 255, 255-72, 255]  # 4Amp
+            stop = [0, 0, 0, 0]
+            if self.verbose:
+                cmd_l, cmd_r = limit_l, limit_r
+                if self.desired_speed < 0:
+                    cmd_l, cmd_r = -cmd_l, -cmd_r
+                elif self.desired_speed == 0:
+                    cmd_l, cmd_r = 0, 0
+                self.enc_debug_arr.append((timestamp.total_seconds(), cmd_l, cmd_r,
+                    self.last_encoders_front_left, self.last_encoders_front_right,
+                    self.last_encoders_rear_left, self.last_encoders_rear_right))
+                self.join_debug_arr.append(self.last_join_angle)
+            if self.desired_speed > 0:
+                if self.last_encoders_front_right is not None:
+                    self.publish('can', CAN_packet(0x31, [0, 0, limit_r//256, limit_r % 256]))
+                    """
+                    if abs(self.last_encoders_front_right) > limit_r:
+                        if abs(self.last_encoders_front_right) - limit_r > limit_brake:
+                            self.publish('can', CAN_packet(0x21, stop))  # brake right front
+                        else:
+                            self.publish('can', CAN_packet(0x11, stop))  # right front
+                    else:
+                        self.publish('can', CAN_packet(0x11, fwd))  # right front"""
+                if self.last_encoders_front_left is not None:
+                    self.publish('can', CAN_packet(0x32, [0, 0, limit_l//256, limit_l % 256]))
+                    """
+                    if abs(self.last_encoders_front_left) > limit_l:
+                        if abs(self.last_encoders_front_left) - limit_l > limit_brake:
+                            self.publish('can', CAN_packet(0x22, stop))  # brake left front
+                        else:
+                            self.publish('can', CAN_packet(0x12, stop))  # left front
+                    else:
+                        self.publish('can', CAN_packet(0x12, fwd))  # left front"""
```

Comment:
What is the plan with the commented out portions inhere?

Reply:
they are obsolete. They were used for current control and we will not return to this.

Reply:
So it might be ok to delete them?

Reply:
yes - this PR is collection of "feature/kloubak" branch, so I would like to fix it there and then cherrypick this fixes. I will add them probably today, thanks.